pkg/datasources/secrets_acceptance_test.go

```diff
@@ -303,3 +215,169 @@ func TestAcc_Secrets_WithGenericString(t *testing.T) {
 		},
 	})
 }
+
+func secretsData(secretResourceName string) string {
+	return fmt.Sprintf(`
+    data "snowflake_secrets" "test" {
+        depends_on = [%s.test]
+    }`, secretResourceName)
+}
+
+func TestAcc_Secrets_Filtering(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	prefix := random.AlphaN(4)
+	idOne := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix)
+	idTwo := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix)
+	idThree := acc.TestClient().Ids.RandomSchemaObjectIdentifierWithPrefix(prefix)
+	idFour := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+
+	integrationId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	_, apiIntegrationCleanup := acc.TestClient().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "test_oauth_client_id", "test_oauth_client_secret").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "first_scope"}, {Scope: "second_scope"}}),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	// ERROR Insufficient privileges to operate on account ... for role BASIC_PRIVILEGES
```

Comment:
let's uncomment it and run on CI (you can also verify it on higher role now)

Reply:
i have not seen any test for `in application` and `in application package` so those have been removed, and altered the `in account` test to use prefix so it doesnt show any other secrets defined outside of this test.